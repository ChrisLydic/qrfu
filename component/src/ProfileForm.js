import React from 'react';
import { Field, reduxForm } from 'redux-form';
import Upload from './lib/Upload';
import { ProfileImage } from './UploadList';

// redux-form example
class ProfileForm extends React.Component {
	constructor(props) {
		super(props);
		this.handleUpload = this.handleUpload.bind(this);
	}

	handleUpload(files) {
		if (files.length > 0) {
			this.props.change('avatar', files[0].url);
		}
	}

	render() {
		const { handleSubmit } = this.props;

		return (
			<form onSubmit={handleSubmit}>
				<Upload ratio={1} filetype='image' multiple={false} uploadlist={ProfileImage} onChange={this.handleUpload}>
					<div>
						<Field name='username' component='input' type="text" placeholder="username" />
						<Field name='password' component='input' type='password' placeholder="password" />
						<button type='submit'>Register</button>
					</div>
				</Upload>

				<Field name='avatar' component='input' type='hidden' />
			</form>
		);
	}
}

ProfileForm = reduxForm({
	form: 'profile',
}) (ProfileForm);

export default ProfileForm;