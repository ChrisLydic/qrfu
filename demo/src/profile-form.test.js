import React from 'react';
import renderer from 'react-test-renderer';
import { combineForms } from 'react-redux-form';
import enzyme, { shallow, mount } from 'enzyme';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import Adapter from 'enzyme-adapter-react-16';
import ProfileForm from './profile-form';

enzyme.configure({ adapter: new Adapter() });

let wrapper;

beforeEach(() => {
    const files = [
        {
            name: 'file-img',
            url: 'http://localhost:3001/',
            uid: '10001010100110',
            type: 'image',
        },
        {
            name: 'file-audio',
            url: 'http://localhost:3001/',
            uid: '00011010100110',
            type: 'audio',
        }
    ];
    const initialState = {
        name: 'fred',
        password: '',
        avatar: '',
    };
    const mockStore = configureStore();
    let store = mockStore(initialState);

    wrapper = shallow(
        <Provider store={store}><ProfileForm /></Provider>,
    ).dive();
});

test('Form displays initial data', () => {
    expect(wrapper.find('#user').getElement().value).toBe('fred')
});