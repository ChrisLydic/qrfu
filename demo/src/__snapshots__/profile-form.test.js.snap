// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form displays 1`] = `undefined`;

exports[`List displays the file links 1`] = `
<ul>
  <li>
    <a
      href="http://localhost:3001/"
    >
      file-img
    </a>
  </li>
  <li>
    <a
      href="http://localhost:3001/"
    >
      file-audio
    </a>
  </li>
</ul>
`;

exports[`List displays uploads in their specific formats 1`] = `
<ul
  className="custom"
>
  <li>
    <a
      href="http://localhost:3001/"
    >
      <img
        alt="preview"
        src="http://localhost:3001/"
      />
      <div>
        file-img
      </div>
    </a>
  </li>
  <li>
    <audio
      controls={true}
      src="http://localhost:3001/"
    />
    <div>
      file-audio
    </div>
  </li>
</ul>
`;

exports[`Profile image displays first file 1`] = `
<img
  alt="avatar"
  className="profilePic"
  src="http://localhost:3001/"
/>
`;

exports[`Profile image is set to default empty tag 1`] = `
<div
  className="profilePic"
/>
`;
